MoltenVK/MoltenVK/Commands/MVKCommandBuffer.h

```diff
@@ -172,6 +172,8 @@ class MVKCommandBuffer : public MVKDispatchableVulkanAPIObject,
 	bool _isReusable;
 	bool _supportsConcurrentExecution;
 	bool _wasExecuted;
+	TimestampCorrelationMarker _timestampCorrelationMarker;
+	std::shared_ptr<MVKTimestampBuffers> _timestampBuffers;
```

Comment:
Using the STL smart pointers is actually quite rare in this project. Maybe we should reconsider that.

Reply:
>Using the STL smart pointers is actually quite rare in this project. Maybe we should reconsider that.

Actually, not rare, but non-existent. 

One of the issues is the use of the `MVKBaseObject` base class to underpin all MoltenVK objects (except POD structs). 

Among other class-hierarchy functionality, that base class is set  up to support a level of memory management. 

Originally this memory management was to include help with Vulkan `VkAllocationCallbacks`, which has yet to be found useful enough to implement. 

But one of the features of this idea that was implemented is the `MVKBaseObject ::destroy()` function, which can be overridden by subclasses with some useful functionality (such as reference counting and deferred deletion). And one of the requirements of **_that_** idea, is that we don't allow classes to be directly deleted via the `delete` operator. 

For better or worse, using STL smart pointers would bypass this management ability.

Reply:
You can provide a custom deleter to `std::shared_pointer` which can call `destroy()`, but I'll see if I can find another alternative.